src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/DelegationValidationStrategy.cs

```diff
@@ -313,18 +327,31 @@ protected bool IsDelegatableColumnNode(FirstNameNode node, TexlBinding binding,
             return true;
         }
 
-        public bool IsValidCallNode(CallNode node, TexlBinding binding, OperationCapabilityMetadata metadata)
+        public virtual bool IsValidCallNode(CallNode node, TexlBinding binding, OperationCapabilityMetadata metadata)
+        {
+            // Functions may have their specific CallNodeDelegationStrategies (i.e. AsType, User)
+            // so, if available, we need to ensure we use their specific delegation strategy.
+            var function = binding.GetInfo(node).Function;
+            return function?.GetCallNodeDelegationStrategy().IsValidCallNode(node, binding, metadata) ?? IsValidCallNodeInternal(node, binding, metadata, false);
+        }
+
+        protected bool IsValidCallNodeInternal(CallNode node, TexlBinding binding, OperationCapabilityMetadata metadata, bool allowNonBlockScopedConstant)
         {
             Contracts.AssertValue(node);
             Contracts.AssertValue(binding);
             Contracts.AssertValue(metadata);
 
-            if (!IsValidNode(node, binding))
+            if (!IsValidAsyncOrImpureNode(node, binding))
             {
                 SuggestDelegationHint(node, binding);
                 return false;
             }
 
+            if (!allowNonBlockScopedConstant && !binding.IsBlockScopedConstant(node))
+            {
+                return false;
+            }
+
```

Comment:
Previously for !binding.IsBlockScopedConstant(node) non-async/pure function, IsValid returned true. I am not sure why that was the case.